### PR TITLE
Add building/linting/testing to make file and fix issues

### PR DESCRIPTION
**Pull request by EliiseS**
In this PR:
- added linting, building, testing and formatting to the make file as in issue #21 
- fixed the issues discovered by building/linting
- format all files

Removed `instance_pools.go` because: 
```bash
golangci-lint run
aws/instance_pools.go:8:27: func `InstancePoolsAPI.init` is unused (unused)
func (a InstancePoolsAPI) init(client DBClient) InstancePoolsAPI {
```
**Questions**
Should I remove instance_pool from azure as well since they are not implemented?

### CODE REVIEW

File: aws/models/ClusterEvent.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package models
 
 type ClusterEvent struct {
-	ClusterID string           `json:"cluster_id,omitempty" url:"cluster_id,omitempty"`
-	Timestamp int64            `json:"timestamp,omitempty" url:"timestamp,omitempty"`
+	ClusterID string            `json:"cluster_id,omitempty" url:"cluster_id,omitempty"`
```
Comment (Azadehkhojandi) [inline aws/models/ClusterEvent.go]: I was wondering if you could remove the space added? Is it possible to revert back this file?

File: Makefile [hunk 2]
```diff
@@ -1,2 +1,16 @@
+all : build test docker-build
```
Comment (Azadehkhojandi) [inline Makefile]: It seems `docker-build` is missing in the make file. I was wondering why do we need docker-build for the SDK?

File: Makefile [hunk 3]
```diff
@@ -1,2 +1,16 @@
...
+
+build:
```
Comment (Azadehkhojandi) [inline Makefile]: I'm not sure if we need to build an SDK, could you please explain the reason behind adding it?

File: Makefile (hunk 2 above)
Comment (EliiseS) [reply Makefile]: That's my bad! I missed taking that out

File: Makefile (hunk 3 above)
Comment (EliiseS) [reply Makefile]: While test and checks catch the build errors, the error message in the build is less cluttered, making it easier to use that to locate where the build is failing:
image
VS 
image

If you think that's not valid enough of a reason, I can remove it as well :)

**Comment by EliiseS** [comment]
The formatting was done using `make fmt` command. Would you like me to still revert them? Previously I also noticed there being difference in the spacing for the same files for azure vs aws.

File: Makefile (hunk 3 above)
Comment (stuartleeks) [reply Makefile]: @Azadehkhojandi and I have just been discussing this and `build` in and SDK project does jar slightly with go packages being shared by source. 

Maybe we could move the `go build` to be the first step in `make checks`?

Comment (EliiseS) [reply Makefile]: Changes made

**Comment by EliiseS** [comment]
@Azadehkhojandi From the discussion between Stuart and you, I've move the build and kept the formatting changes

**Comment by Azadehkhojandi** [comment]
Thanks for your PR @EliiseS , When I switched to your branch for the last final check, the dev container is broken

> [2020-03-12T10:52:43.617Z] [PID 19112] [173282 ms] ERROR: Service 'docker-in-docker' failed to build: The command '/bin/sh -c apt-get update     && apt-get -y install --no-install-recommends apt-utils 2>&1     && apt-get -y install git procps lsb-release     && apt-get install vim -y     && go get -x -d github.com/stamblerre/gocode 2>&1     && go build -o gocode-gomod github.com/stamblerre/gocode     && mv gocode-gomod $GOPATH/bin/     && go get -u -v         github.com/mdempsky/gocode         github.com/uudashr/gopkgs/cmd/gopkgs         github.com/ramya-rao-a/go-outline         github.com/acroca/go-symbols         github.com/godoctor/godoctor         golang.org/x/tools/cmd/guru         golang.org/x/tools/cmd/gorename         github.com/rogpeppe/godef         github.com/zmb3/gogetdoc         github.com/haya14busa/goplay/cmd/goplay         github.com/sqs/goreturns         github.com/josharian/impl         github.com/davidrjenni/reftools/cmd/fillstruct         github.com/fatih/gomodifytags         github.com/cweill/gotests/...         g
> [2020-03-12T10:52:43.662Z] [PID 19112] [173327 ms] 
> olang.org/x/tools/cmd/goimports         golang.org/x/lint/golint         golang.org/x/tools/cmd/gopls         github.com/alecthomas/gometalinter         honnef.co/go/tools/...         github.com/golangci/golangci-lint/cmd/golangci-lint         github.com/mgechev/revive         github.com/derekparker/delve/cmd/dlv 2>&1     && apt-get autoremove -y     && apt-get clean -y     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 1
> [2020-03-12T10:52:43.739Z] [PID 19112] [173404 ms] Failed: docker-compose up

Could you please quickly check it?

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -70,3 +60,7 @@ COPY ./Makefile ./
 
 RUN mkdir -p /go/src/github.com/xinsnake/databricks-sdk-golang
 ENV SHELL /bin/bash
+
+
+# # Install golangci-lint
```
Comment (Azadehkhojandi) [inline .devcontainer/Dockerfile]: Please remove comment out lines

Comment (EliiseS) [reply .devcontainer/Dockerfile]: Ah good catch!

**Pull request merged**